src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -14,100 +16,129 @@
+      hashtable.put(
+          "perception-term", String.parseString(latestRating.getProperty("perception-term")));
+      hashtable.put("hours", String.parseString(latestRating.getProperty("hours")));
+      hashtable.put("difficulty", String.parseString(latestRating.getProperty(difficulty)));
+      hashtable.put("comments-professor", latestRating.getProperty(comments - professor));
+      hashtable.put("perception-professor", latestRating.getProperty(perception - professor));
     }
 
     // Send JSON string.
-    String jsonVersionCommentsList = new Gson().toJson(commentsList);
-    commentsList.clear();
+    String jsonVersionHashtable = new Gson().toJson(hashtable);
+    ratingQueryList.clear();
     response.setContentType("application/json;");
-    response.getWriter().println(jsonVersionCommentsList);
+    response.getWriter().println(jsonVersionHashtable);
   }
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
     // Get written feedback.
-    String classFeedback = request.getParameter("class-input");
+    addTermRating(request);
+    response.setContentType("text/html; charset=UTF-8");
+    response.setCharacterEncoding("UTF-8");
+    response.sendRedirect("/index.html");
+  }
+
+  public void addTermRating(HttpServletRequest request) {
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    String termFeedback = request.getParameter("term-input");
+    Long termRating = Integer.parseInt(request.getParameter("rating-term"));
+    Long workHours = Integer.parseInt(request.getParameter("hoursOfWork"));
+    Long difficulty = Integer.parseInt(request.getParameter("difficulty"));
     String professorFeedback = request.getParameter("prof-input");
-    // Get class/professor ratings.
-    String classRating = request.getParameter("rating-class");
-    String professorRating = request.getParameter("rating-professor");
+    Long professorRating = Integer.parseInt(request.getParameter("rating-professor"));
     boolean translateToEnglish = Boolean.parseBoolean(request.getParameter("languages"));
 
     if (translateToEnglish) {
-      Translate translateService = TranslateOptions.getDefaultInstance().getService();
-      Translation translationClassFeedback =
-          translateService.translate(classFeedback, Translate.TranslateOption.targetLanguage("en"));
-      Translation translationProfessorFeedback =
-          translateService.translate(
-              professorFeedback, Translate.TranslateOption.targetLanguage("en"));
-      String translatedClassFeedback = translationClassFeedback.getTranslatedText();
-      String translatedProfessorFeedback = translationProfessorFeedback.getTranslatedText();
-      classFeedback = translatedClassFeedback;
-      professorFeedback = translatedProfessorFeedback;
+      termFeedback = translateFeedback(termFeedback);
+      professorFeedback = translateFeedback(professorFeedback);
```

Comment:
make sure this is actually changing in your tests, just bc you're modifying the same variable instead of making a new one it looks like

Reply:
When I deploy it shows that this is working thankfully!